core/engine/src/builtins/temporal/duration/mod.rs

```diff
@@ -929,94 +928,94 @@ pub(crate) fn to_temporal_partial_duration(
     };
 
     // 2. Let result be a new partial Duration Record with each field set to undefined.
-    let mut result = InnerDuration::partial();
+    let mut result = PartialDuration::default();
 
     // 3. NOTE: The following steps read properties and perform independent validation in alphabetical order.
     // 4. Let days be ? Get(temporalDurationLike, "days").
     let days = unknown_object.get(js_str!("days"), context)?;
     if !days.is_undefined() {
         // 5. If days is not undefined, set result.[[Days]] to ? ToIntegerIfIntegral(days).
-        result.set_days(f64::from(to_integer_if_integral(&days, context)?));
+        result.days = Some(FiniteF64::from(to_integer_if_integral(&days, context)?));
     }
 
     // 6. Let hours be ? Get(temporalDurationLike, "hours").
     let hours = unknown_object.get(js_str!("hours"), context)?;
     // 7. If hours is not undefined, set result.[[Hours]] to ? ToIntegerIfIntegral(hours).
     if !hours.is_undefined() {
-        result.set_hours(f64::from(to_integer_if_integral(&hours, context)?));
+        result.hours = Some(FiniteF64::from(to_integer_if_integral(&hours, context)?));
     }
 
     // 8. Let microseconds be ? Get(temporalDurationLike, "microseconds").
     let microseconds = unknown_object.get(js_str!("microseconds"), context)?;
     // 9. If microseconds is not undefined, set result.[[Microseconds]] to ? ToIntegerIfIntegral(microseconds).
     if !microseconds.is_undefined() {
-        result.set_microseconds(f64::from(to_integer_if_integral(&microseconds, context)?));
+        result.microseconds = Some(FiniteF64::from(to_integer_if_integral(
+            &microseconds,
+            context,
+        )?));
     }
 
     // 10. Let milliseconds be ? Get(temporalDurationLike, "milliseconds").
     let milliseconds = unknown_object.get(js_str!("milliseconds"), context)?;
     // 11. If milliseconds is not undefined, set result.[[Milliseconds]] to ? ToIntegerIfIntegral(milliseconds).
     if !milliseconds.is_undefined() {
-        result.set_milliseconds(f64::from(to_integer_if_integral(&milliseconds, context)?));
+        result.milliseconds = Some(FiniteF64::from(to_integer_if_integral(
+            &milliseconds,
+            context,
+        )?));
     }
 
     // 12. Let minutes be ? Get(temporalDurationLike, "minutes").
     let minutes = unknown_object.get(js_str!("minutes"), context)?;
     // 13. If minutes is not undefined, set result.[[Minutes]] to ? ToIntegerIfIntegral(minutes).
     if !minutes.is_undefined() {
-        result.set_minutes(f64::from(to_integer_if_integral(&minutes, context)?));
+        result.minutes = Some(FiniteF64::from(to_integer_if_integral(&minutes, context)?));
```

Comment:
thought: It might be nice if we had a `JsValue::map` that acted similar to `Option::map`, where Undefined is None.

Reply:
Im still wondering how to do this in a clean way, was you thinking `result.minutes = minutes.map(|minutes| FiniteF64::from(to_integer_if_integral(&minutes, context)?)` ?